ConfigurationSystem/Client/Helpers/Registry.py

```diff
@@ -223,45 +292,57 @@ def getUsersInVO(vo, defaultValue=None):
 
       :return: list
   """
+  users = []
   result = getGroupsForVO(vo)
-  if not result['OK'] or not result['Value']:
-    return [] if defaultValue is None else defaultValue
-  groups = result['Value']
+  if result['OK'] and result['Value']:
+    for group in result['Value']:
+      users += getUsersInGroup(group)
 
-  userList = []
-  for group in groups:
-    userList += getUsersInGroup(group)
-  return userList
+  users.sort()
+  return list(set(users)) or [] if defaultValue is None else defaultValue
 
 
-def getDNsInVO(vo):
-  """ Get all DNs that have a VO users
+def getDNsInGroup(group, checkStatus=False):
+  """ Find user DNs for DIRAC group
 
-      :param str vo: VO name
+      :param str group: group name
+      :param bool checkStatus: don't add suspended DNs
 
       :return: list
   """
-  DNs = []
-  for user in getUsersInVO(vo):
-    result = getDNForUsername(user)
-    if result['OK']:
-      DNs.extend(result['Value'])
-  return DNs
-
+  vomsData = {}
+  vo = getGroupOption(group, 'VO')
 
-def getDNsInGroup(groupName):
-  """ Find all DNs  for DIRAC group
-
-      :param str groupName: group name
+  # Get VOMS information for VO, if it's VOMS VO
+  result = getVOsWithVOMS([vo])
+  if not result['OK']:
+    return result
+  if result['Value']:
+    result = getVOMSInfo(vo=vo)
+    if not result['OK']:
+      return result
+    vomsData = result['Value']
 
-      :return: list
-  """
   DNs = []
-  for user in getUsersInGroup(groupName):
-    result = getDNForUsername(user)
-    if result['OK']:
-      DNs.extend(result['Value'])
-  return DNs
+  for username in getGroupOption(group, 'Users', []):
+    if checkStatus and vo in getUserOption(username, 'Suspended', []):
+      continue
+    result = getDNsForUsername(username)
+    if not result['OK']:
+      return result
+    userDNs = result['Value']
+    if vomsData.get(vo) and vomsData[vo]['OK']:
+      voData = vomsData[vo]['Value']
+      role = getGroupOption(group, 'VOMSRole')
+      for dn in userDNs:
+        if dn in voData:
+          if not checkStatus or not voData[dn]['Suspended']:
+            if not role or role in voData[dn]['ActuelRoles' if checkStatus else 'VOMSRoles']:
```

Comment:
Assuming there isn't a backwards compatibility reason not to change this everywhere:
```suggestion
            if not role or role in voData[dn]['ActiveRoles' if checkStatus else 'VOMSRoles']:
```

Reply:
fixed